### DISCUSSION TITLE
Order matters for if expressions - cannot add field..was already used error

### DISCUSSION THREAD

**Discussion by cueckoo** [cue-lang/cue#1075] [Q&A] [answered]
_Originally opened by @KostasKgr in https://github.com/cuelang/cue/discussions/1075_

Hello all, I wasn't sure if the right place for this is here, or in slack or in issues. Let me know!

I've been trying to model our multi customer configuration and whilst trying it I saw a behaviour that I didn't expect.

I'm trying to understand if it is an issue or not, but it definitely breaks composition of different files for the approach I was trying.

So to showcase this, i have two files:

feature_definitions.cue
```
package config

features: {
    myFeature: *false | bool
}

if features.myFeature {
    myService: {
        config: {
            replicas: 4
        }
    }    
}
```
regional_configs.cue
```
package config

region: "US"
if region == "US" {
    features: myFeature: true
}
```

If I include feature definitions before regional configs then I get an error, If i do it the other way around I get the intended outcome.
```
15:41 $ cue eval -c feature_definitions.cue regional_configs.cue 
cannot add field myFeature: was already used:
    ./regional_configs.cue:5:15
```

```
15:46 $ cue eval -c regional_configs.cue feature_definitions.cue 
region: "US"
features: {
    myFeature: true
}
myService: {
    config: {
        replicas: 4
    }
}
```

Does using expressions limit splitting out the definitions across multiple files and thus in random composition?
Is this an issue or the intended outcome?

Thank you!

**Comment by cueckoo** [comment, marked as answer]
_Original reply by @verdverm in https://github.com/cuelang/cue/discussions/1075#discussioncomment-944907_

Given CUE is order agnostic, this does seem like a bug to me. Note this behavior is the same across eval, export, and def.

Can you open an issue following the guidelines here: https://github.com/cuelang/cue/wiki/Creating-test-or-performance-reproducers

**Reply by KostasKgr** [reply]
Raised https://github.com/cue-lang/cue/issues/1083

How can I accept this as the answer?
